logging/interface.go

```diff
@@ -0,0 +1,43 @@
+// Package logging defines a logging interface for quic-go.
+// This package should not be considered stable
+package logging
+
+import (
+	"net"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/internal/congestion"
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/wire"
+)
+
+type Tracer interface {
+	TracerForServer(odcid protocol.ConnectionID) ConnectionTracer
+	TracerForClient(odcid protocol.ConnectionID) ConnectionTracer
+}
+
+// A ConnectionTracer records events.
+type ConnectionTracer interface {
```

Comment:
To make this useful, we'll have to make sure that all function parameters don't use any internal types. For most types, we can get away with defining type alias, e.g.
```go
type MaxDataFrame = wire.MaxDataFrame
```
Not sure if the fact the `MaxDataFrame` uses a `protocol.ByteCount` (which is also an internal type) will cause any problems here.

For some types, we might want to redefine the type. For example, we don't want to pass the contents of STREAM and CRYPTO frames to the logger:
```go
type StreamFrame struct {
    Offset protocol.ByteCount
    StreamID protocol.StreamID
    Length int
}
```

Suggestion:
We'll create a new package *logging/internal/to_logging*, that defines conversion for these types:
```go
func ConvertFrame(*wire.Frame) *logging.Frame
```

Reply:
SGTM. Maybe we can reuse this conversion in the `LogFrame` function?

Reply:
Actually I wonder whether the Logger can eventually be yet another tracer… Probably not?

Reply:
That would be an option. We could then define a `String() string` method on the `logging.Frame`, and get rid of the massive `switch` statement in `LogFrame`.

Reply:
Maybe you don't even need that String() and can just use the fmt print?

Reply:
That would print the contents of STREAM and CRYPTO frames.

Reply:
But the LogFrame wouldn't include those, and instead have useful metadata (len, etc)? I guess the downside would be that we don't get to format fields usefully (hex vs dec e.g.), but not sure whether that's worth the complexity?